File: src/main/java/aurocosh/divinefavor/common/custom_data/player/data/spell/grudge/GrudgeDataSerializer.java
package aurocosh.divinefavor.common.custom_data.player.data.spell.grudge;

import aurocosh.divinefavor.common.lib.interfaces.INbtSerializer;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.util.ResourceLocation;

// Handles the actual read/write of the nbt.
public class GrudgeDataSerializer implements INbtSerializer<GrudgeData> {
    private static String TAG_GRUDGE = "Grudge";

    @Override
    public void serialize(NBTTagCompound nbt, GrudgeData instance) {
        nbt.setString(TAG_GRUDGE, instance.getMobTypeId().toString());
    }

    @Override
    public void deserialize(NBTTagCompound nbt, GrudgeData instance) {
        String value = nbt.getString(TAG_GRUDGE);
        instance.setMobTypeId(new ResourceLocation(value));
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/custom_data/player/data/spell/grudge/GrudgeDataSerializer.kt
package aurocosh.divinefavor.common.custom_data.player.data.spell.grudge

import aurocosh.divinefavor.common.lib.interfaces.INbtSerializer
import net.minecraft.nbt.NBTTagCompound
import net.minecraft.util.ResourceLocation

// Handles the actual read/write of the nbt.
class GrudgeDataSerializer : INbtSerializer<GrudgeData> {

    override fun serialize(nbt: NBTTagCompound, instance: GrudgeData) {
        nbt.setString(TAG_GRUDGE, instance.mobTypeId.toString())
    }

    override fun deserialize(nbt: NBTTagCompound, instance: GrudgeData) {
        val value = nbt.getString(TAG_GRUDGE)
        instance.mobTypeId = ResourceLocation(value)
    }

    companion object {
        private const val TAG_GRUDGE = "Grudge"
    }
}
